pennylane/tape/tape.py

```diff
@@ -1036,12 +1037,70 @@ def get_depth(self):
 
         >>> tape.get_depth()
         4
+
         """
+
+        warnings.warn(
+            "``tape.get_depth`` will be deprecated after v0.16 "
+            "Please use the more general ``tape.specs`` instead.",
+            UserWarning,
+        )
+
         if self._depth is None:
             self._depth = self.graph.get_depth()
 
         return self._depth
 
+    @property
+    def specs(self):
+        """Resource requirements of a quantum circuit.
+
+        Returns:
+            dict[str, Union[defaultdict,int]]: how many times constituent operations are applied
+
+        **Example**
+
+        .. code-block:: python3
+
+            with qml.tape.QuantumTape() as tape:
+                qml.Hadamard(wires=0)
+                qml.RZ(0.26, wires=1)
+                qml.CNOT(wires=[1, 0])
+                qml.Rot(1.8, -2.7, 0.2, wires=0)
+                qml.Hadamard(wires=1)
+                qml.CNOT(wires=[0, 1])
+                qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        Asking for the resources produces a dictionary as shown below:
+
+        >>> tape.specs['by_size']
+        defaultdict(int, {1: 4, 2: 2})
+        >>> tape.specs['by_name']
+        defaultdict(int, {'Hadamard': 2, 'RZ': 1, 'CNOT': 2, 'Rot': 1})
+
+        As `defaultdict` objects, any key not present in the dictionary returns 0.
+
+        >>> tape.specs['by_name']['RX']
+        0
+
+        """
+        if self._specs is None:
+            self._specs = {"by_size": defaultdict(int), "by_name": defaultdict(int)}
+
+            for op in self.operations:
+                # don't use op.num_wires to allow for flexible gate classes like hermitian
```

Comment:
(minor) seems like ops are processed here, would expect Hermitian to be among `self.observables`
```suggestion
                # don't use op.num_wires to allow for flexible gate classes like QubitUnitary
```